### ISSUE TITLE
JupyterViz / Solara visualisation doesn't work in JetBrains PyCharm/DataSpell Jupyter notebooks

### ISSUE THREAD

**Issue by EwoutH** [projectmesa/mesa#1853]
Currently, JupyterViz / Solara visualisation doesn't work in JetBrains PyCharm/DataSpell Jupyter notebooks. This isn't a Mesa problem, but it's good to be aware of it.

![Screenshot_176](https://github.com/projectmesa/mesa/assets/15776622/94a53996-f8bc-402b-9022-d418c8c21680)

- Issue at JetBrains: https://youtrack.jetbrains.com/issue/PY-63760
- Discussion at Solara: https://github.com/widgetti/solara/discussions/354

**Comment by rht** [comment]
Is there a way to detect via `sys.argv[0]` that it is run on JetBrains PyCharm/DataSpell Jupyter?

**Comment by rht** [comment]
If so, then we can raise a more informative error message pointing to this issue.

**Comment by EwoutH** [comment]
`sys.argv[0]` gives

```
'C:\\Users\\Ewout\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ipykernel_launcher.py'
```
Is that useful?

**Comment by rht** [comment]
It doesn't contain information on whether it is run from JetBrains PyCharm/DataSpell. There could already be an env variable set (see https://youtrack.jetbrains.com/issue/PY-4853), but it's hard for me to try without access to the IDE.

**Comment by EwoutH** [comment]
Took some time, but it looks like it's fixed in on their PyCharm internally. That would mean with the next release it should be fixed!

- https://youtrack.jetbrains.com/issue/PY-63760

**Comment by EwoutH** [comment]
I can confirm that this issue is fixed starting from PyCharm 2023.3.3 RC!
